### ISSUE TITLE
Simultaneous utilization of same resource by multiple tasks

### ISSUE THREAD

**Issue by aayushm10001** [traherom/simple-scheduler#3]
Hi, thanks for this library! It's pretty useful in visualizing how computer tasks can be scheduled.

I think I have found a bug. In the files attached (CSV+SVG), you can see InR3 and OutR2 both utilizing the resource 'EDMA' at the same time. I have changed the library to schedule on the weekends too, but the issue also existed with the library as is.

[example_schedule.zip](https://github.com/traherom/simple-scheduler/files/5916580/example_schedule.zip)

**Comment by traherom** [comment]
Good catch! Unfortunately I don't have time to investigate this, but I'll happily merge any PRs I get.

**Comment by itopaloglu83** [comment]
A possible explanation for why the resource was double booked.

It looks like OutR2 was scheduled first and acquired time on EDMA resource. When scheduling a task, the code only checks for the resource to be available on the task start date. In this case InR3 started on a free date but its end happens to overlap with another task.

One way to resolve the problem is changing the order of tasks so that InR3 gets scheduled before OutR2. A permanent solution would require checking the resource availability across the task duration. I might submit a PR sometime if I can.

https://github.com/traherom/simple-scheduler/blob/589fde310dd259e9466104cb63e99b0b92ced307/projectscheduler/gantt.py#L528

**Comment by itopaloglu83** [comment]
For future reference, [python-gantt](https://pypi.org/project/python-gantt/#files) has a similar feature. 
``` Python
  def check_conflicts_between_task_and_resources_vacations(self):
      """
      Displays a warning for each conflict between tasks and vacation of
      resources affected to the task

      And returns a dictionnary for resource vacation conflicts
      """
      conflicts = []
      if self.get_resources() is None:
          return conflicts
      for r in self.get_resources():
          cday = self.start_date()
          while cday <= self.end_date():
              if cday.weekday() not in _not_worked_days() and not r.is_available(cday):
                  conflicts.append({'resource':r.name,'date':cday, 'task':self.name})
                  __LOG__.warning('** Caution resource "{0}" is affected on task "{2}" during vacations on day {1}'.format(r.name, cday, self.fullname))
              cday += datetime.timedelta(days=1)
```
